<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>Server</class>
 <widget class="QMainWindow" name="Server">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1236</width>
    <height>837</height>
   </rect>
  </property>
  <property name="sizePolicy">
   <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
    <horstretch>0</horstretch>
    <verstretch>0</verstretch>
   </sizepolicy>
  </property>
  <property name="font">
   <font>
    <pointsize>14</pointsize>
   </font>
  </property>
  <property name="windowTitle">
   <string>通讯</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QHBoxLayout" name="horizontalLayout">
    <item>
     <widget class="QTabWidget" name="tabWidget">
      <property name="currentIndex">
       <number>0</number>
      </property>
      <widget class="QWidget" name="tab_2">
       <attribute name="title">
        <string>PLC</string>
       </attribute>
       <layout class="QVBoxLayout" name="verticalLayout_7">
        <item>
         <layout class="QGridLayout" name="gridLayout_6">
          <item row="0" column="0">
           <layout class="QVBoxLayout" name="verticalLayout_5">
            <item>
             <widget class="QLabel" name="label">
              <property name="layoutDirection">
               <enum>Qt::LeftToRight</enum>
              </property>
              <property name="text">
               <string>ModbusTcp</string>
              </property>
              <property name="textFormat">
               <enum>Qt::AutoText</enum>
              </property>
              <property name="alignment">
               <set>Qt::AlignCenter</set>
              </property>
             </widget>
            </item>
            <item>
             <layout class="QVBoxLayout" name="verticalLayout_4">
              <item>
               <widget class="QLabel" name="label_9">
                <property name="text">
                 <string>返回显示</string>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QPlainTextEdit" name="te_show"/>
              </item>
             </layout>
            </item>
           </layout>
          </item>
          <item row="0" column="1">
           <layout class="QVBoxLayout" name="verticalLayout_3">
            <item>
             <widget class="QWidget" name="widget" native="true">
              <layout class="QGridLayout" name="gridLayout_5">
               <item row="0" column="0">
                <widget class="QLabel" name="label_8">
                 <property name="text">
                  <string>IP</string>
                 </property>
                </widget>
               </item>
               <item row="0" column="1">
                <widget class="QLineEdit" name="le_ip"/>
               </item>
               <item row="0" column="2">
                <widget class="QLabel" name="lb_state">
                 <property name="text">
                  <string>连接状态</string>
                 </property>
                </widget>
               </item>
               <item row="1" column="0">
                <widget class="QLabel" name="label_7">
                 <property name="text">
                  <string>Port</string>
                 </property>
                </widget>
               </item>
               <item row="1" column="1">
                <widget class="QLineEdit" name="le_port"/>
               </item>
               <item row="1" column="2">
                <widget class="QPushButton" name="pb_connect">
                 <property name="text">
                  <string>连接</string>
                 </property>
                </widget>
               </item>
              </layout>
             </widget>
            </item>
            <item>
             <widget class="QGroupBox" name="groupBox">
              <property name="title">
               <string>线圈寄存器</string>
              </property>
              <layout class="QGridLayout" name="gridLayout_3">
               <item row="2" column="0">
                <widget class="QLabel" name="label_11">
                 <property name="text">
                  <string>地址</string>
                 </property>
                </widget>
               </item>
               <item row="4" column="2">
                <widget class="QPushButton" name="pb_writeM">
                 <property name="text">
                  <string>写入</string>
                 </property>
                </widget>
               </item>
               <item row="4" column="0">
                <widget class="QLabel" name="label_12">
                 <property name="text">
                  <string>数据</string>
                 </property>
                </widget>
               </item>
               <item row="2" column="2">
                <widget class="QPushButton" name="pb_readM">
                 <property name="text">
                  <string>读取</string>
                 </property>
                </widget>
               </item>
               <item row="4" column="1">
                <widget class="QLineEdit" name="le_dataM"/>
               </item>
               <item row="2" column="1">
                <widget class="QLineEdit" name="le_addressM"/>
               </item>
              </layout>
             </widget>
            </item>
            <item>
             <widget class="QGroupBox" name="groupBox_2">
              <property name="title">
               <string>保持寄存器</string>
              </property>
              <layout class="QGridLayout" name="gridLayout_4">
               <item row="3" column="1">
                <widget class="QLineEdit" name="le_dataD"/>
               </item>
               <item row="3" column="2">
                <widget class="QPushButton" name="pb_writeD">
                 <property name="text">
                  <string>写入</string>
                 </property>
                </widget>
               </item>
               <item row="3" column="0">
                <widget class="QLabel" name="label_14">
                 <property name="text">
                  <string>数据</string>
                 </property>
                </widget>
               </item>
               <item row="2" column="1">
                <widget class="QLineEdit" name="le_addressD"/>
               </item>
               <item row="2" column="0">
                <widget class="QLabel" name="label_13">
                 <property name="text">
                  <string>地址</string>
                 </property>
                </widget>
               </item>
               <item row="2" column="2">
                <widget class="QPushButton" name="pb_readD">
                 <property name="text">
                  <string>读取</string>
                 </property>
                </widget>
               </item>
               <item row="1" column="1">
                <widget class="QPushButton" name="pb_stop">
                 <property name="text">
                  <string>停止</string>
                 </property>
                </widget>
               </item>
               <item row="0" column="1">
                <widget class="QPushButton" name="pb_start">
                 <property name="text">
                  <string>开始</string>
                 </property>
                </widget>
               </item>
              </layout>
             </widget>
            </item>
           </layout>
          </item>
         </layout>
        </item>
       </layout>
      </widget>
      <widget class="QWidget" name="tab">
       <attribute name="title">
        <string>机械臂</string>
       </attribute>
       <layout class="QGridLayout" name="gridLayout_2">
        <item row="0" column="0">
         <layout class="QVBoxLayout" name="verticalLayout">
          <item>
           <widget class="QPushButton" name="pushButton_StartToListen">
            <property name="styleSheet">
             <string notr="true"/>
            </property>
            <property name="text">
             <string>监听</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QLabel" name="label_hostName">
            <property name="text">
             <string>hostName:</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QLabel" name="label_2">
            <property name="text">
             <string>本机IP：</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QLineEdit" name="lineEdit_Local_IP"/>
          </item>
          <item>
           <widget class="QLabel" name="label_3">
            <property name="text">
             <string>端口：</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QLineEdit" name="lineEdit_Local_Port"/>
          </item>
          <item>
           <widget class="QLabel" name="label_6">
            <property name="text">
             <string>IP列表：</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QListWidget" name="listWidget_IP"/>
          </item>
          <item>
           <widget class="QPushButton" name="pushButton_Send">
            <property name="text">
             <string>发送</string>
            </property>
           </widget>
          </item>
         </layout>
        </item>
        <item row="0" column="1">
         <layout class="QVBoxLayout" name="verticalLayout_2">
          <item>
           <widget class="QLabel" name="label_4">
            <property name="text">
             <string>接收：</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QPlainTextEdit" name="plainTextEdit_Recv_Msg"/>
          </item>
          <item>
           <widget class="QLabel" name="label_5">
            <property name="text">
             <string>发送：</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QPlainTextEdit" name="plainTextEdit_Send_Msg"/>
          </item>
         </layout>
        </item>
       </layout>
      </widget>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>1236</width>
     <height>26</height>
    </rect>
   </property>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
 </widget>
 <resources/>
 <connections/>
</ui>
